lua/packer/plugin_types/git.lua

```diff
@@ -128,6 +140,31 @@ local handle_checkouts = function(plugin, dest, disp)
   end)
 end
 
+local get_rev = function(plugin)
+  local plugin_name = util.get_plugin_full_name(plugin)
+
+  local rev_cmd = config.exec_cmd .. config.subcommands.get_rev
+
+  return async(function()
+    local rev = await(
+      jobs.run(
+        rev_cmd,
+        { cwd = plugin.install_path, options = { env = git.job_env }, capture_output = true }
+      ))
+    :map_ok(function (ok)
+        local _, r = next(ok.output.data.stdout)
+        return r
+      end)
+    :map_err(function(err)
+        local _, msg = fmt("%s: %s",plugin_name,next(err.output.data.stderr))
+        error(msg)
+        return ""
```

Comment:
This will just set `rev.err = ''` - we should probably set it to the error message string. Also, we'll never reach this code anyway because of the `error` call above it.

I'd suggest removing the `error` call and returning `rev` from `get_rev` (instead of `rev.ok`). Then callers of `get_rev` can decide how to handle any potential errors.